clients/drcachesim/tools/view.cpp

```diff
@@ -205,6 +157,78 @@ view_t::process_memref(const memref_t &memref)
     return parallel_shard_memref(serial_stream_, memref);
 }
 
+bool
+view_t::init_decode_cache()
+{
+    assert(decode_cache_ == nullptr);
+    decode_cache_ =
+        std::unique_ptr<decode_cache_t<disasm_info_t>>(new decode_cache_t<disasm_info_t>(
+            dcontext_.dcontext,
+            // We will decode the instr_t ourselves.
+            /*include_decoded_instr=*/false,
+            /*persist_decoded_instrs=*/false, knob_verbose_));
+    if (TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, filetype_)) {
+        error_string_ = decode_cache_->init(static_cast<offline_file_type_t>(filetype_));
+    } else if (!module_file_path_.empty()) {
+        error_string_ = decode_cache_->init(static_cast<offline_file_type_t>(filetype_),
+                                            module_file_path_, knob_alt_module_dir_);
+    } else {
+        // Continue but omit disassembly to support cases where binaries are
+        // not available and OFFLINE_FILE_TYPE_ENCODINGS is not present.
+        decode_cache_.reset(nullptr);
+    }
+    return error_string_.empty();
+}
+
+bool
+view_t::init_from_filetype()
+{
+    if (init_from_filetype_done_) {
```

Comment:
This is safe because this tool only supports serial mode?

Reply:
Yes.

See https://github.com/DynamoRIO/dynamorio/blob/f007194c734b769c7af69df332c9d36583b97281/clients/drcachesim/tools/view.cpp#L152

Also see existing similar treatment of all other view_t data members https://github.com/DynamoRIO/dynamorio/blob/f007194c734b769c7af69df332c9d36583b97281/clients/drcachesim/tools/view.h#L144

https://github.com/DynamoRIO/dynamorio/blob/f007194c734b769c7af69df332c9d36583b97281/clients/drcachesim/tools/view.cpp#L218